### PR TITLE
tgbot: add version 1.8

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **tgbot/1.8**

#### Motivation
There are several improvements since 1.7.2.
tgbot has required C++17 since 1.7.3.

#### Details
https://github.com/reo7sp/tgbot-cpp/compare/v1.7.2...v1.8

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by valgur — APPROVED** [review]

**Review by valgur — APPROVED** [review]

**Review by ErniGH — APPROVED** [review]
Thanks!

**Review by AbrilRBS — APPROVED** [review]
Thanks @toge, sorry for the delay!

**Review by ErniGH — DISMISSED** [review]
Thanks!

**Review by AbrilRBS — APPROVED** [review]

**Review by AbrilRBS — DISMISSED** [review]

File: recipes/tgbot/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
+project(test_package LANGUAGES CXX)
 
 find_package(tgbot REQUIRED CONFIG)
 
 add_executable(${PROJECT_NAME} test_package.cpp)
-set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
-target_link_libraries(${PROJECT_NAME} tgbot::tgbot)
+target_link_libraries(${PROJECT_NAME} PRIVATE tgbot::tgbot)
+if(tvbot_VERSION VERSION_LESS "1.7.3")
+    target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_11)
```
Comment (AbrilRBS) [inline recipes/tgbot/all/test_package/CMakeLists.txt:9]:
```suggestion
    target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_14)
```

**Review by AbrilRBS — APPROVED** [review]

**Review by valgur — APPROVED** [review]

**Review by ErniGH — APPROVED** [review]
Thanks!

**Pull request merged**